tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -247,6 +248,23 @@ def with_poetry(context: PipelineContext) -> Container:
     return poetry_with_cache
 
 
+def with_pipx(context: PipelineContext) -> Container:
+    """Installs pipx in a python environment.
```

Comment:
Why do we need pipx?

Reply:
Pipx is a handy shortcut to either running a python / poetry module

```bash
# from anywhere
pipx run <path-to-module> metadata-service validate metadata <file-path>
```

or

installing it globally
```bash
pipx install <path-to-module>

# you can now call this anywhere in the dir tree
metadata-service validate metadata <file-path>
```

This means it should be alot simpler for us to isolate our cli tools while still being able to run them easily from anywhere